### DISCUSSION TITLE
Can I build LLVM with Emscripten?

### DISCUSSION THREAD

**Discussion by ricmoo** [emscripten-core/emscripten#20234] [Q&A]
I'm trying to build a browser-based C-to-RISCV compiler tool, but trying to search the internet with terms like "compile llvm with emcripten" gets confused (as emscripten is pretty tightly related to LLVM ;)).

Does anyone know of any existing project (I did fine llvm.js, but it seems 11 years old) or the simplest place to start?

Thanks! :)

RicMoo

**Comment by ricmoo** [comment]
Also, I'm also interested in building gcc with emscripten. I have considered creating an emscripten "host" for crosstool-NG, but haven't had much luck with it picking up my overrides for the compiler.

**Comment by sbc100** [comment]
There is a bunch of info to be found on this here: https://github.com/emscripten-core/emscripten/issues/6432
